analyzers/src/SonarAnalyzer.CSharp/Rules/DatabasePasswordsShouldBeSecure.cs

```diff
@@ -69,6 +72,38 @@ protected override void Initialize(TrackerInput input)
             inv.Track(input, inv.MatchMethod(trackedInvocations), HasEmptyPasswordArgument());
         }
 
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            base.Initialize(context);
+            context.RegisterCompilationAction(c => CheckWebConfig(context, c));
+        }
+
+        private void CheckWebConfig(SonarAnalysisContext context, CompilationAnalysisContext c)
+        {
+            foreach (var fullPath in context.GetWebConfig(c))
+            {
+                var webConfig = File.ReadAllText(fullPath);
+                if (webConfig.Contains("<connectionStrings>") && XmlHelper.ParseXDocument(webConfig) is { } doc)
+                {
+                    ReportEmptyPassword(doc, fullPath, c);
+                }
+            }
+        }
+
+        private void ReportEmptyPassword(XDocument doc, string webConfigPath, CompilationAnalysisContext c)
+        {
+            foreach (var addAttribute in doc.XPathSelectElements("configuration/connectionStrings/add"))
+            {
+                if (addAttribute.Attribute("connectionString") is { } connectionString
```

Comment:
We can cover also this with an incomplete `add` node. Eg. `<add name="Name"
       providerName="System.Data.ProviderName" />`